src/lib/y2users/linux/users_writer.rb

```diff
@@ -0,0 +1,366 @@
+
+        Yast::Execute.on_target!(CHAGE, *options, user.name)
+      rescue Cheetah::ExecutionFailed => e
+        issues << Y2Issues::Issue.new(
+          # TRANSLATORS: %s is a placeholder for a username
+          format(_("Error setting the properties of the password for '%s'"), user.name)
+        )
+        log.error("Error setting password attributes for '#{user.name}' - #{e.message}")
+      end
+
+      # Attributes to modify using `usermod`
+      USERMOD_ATTRS = [:gid, :home, :shell, :gecos].freeze
+
+      # Edits the user
+      #
+      # @param new_user [User] User containing the updated information
+      # @param old_user [User] Original user
+      # @param issues [Y2Issues::List] a collection for adding an issue if something goes wrong
+      def edit_user(new_user, old_user, issues)
+        usermod_changes = USERMOD_ATTRS.any? do |attr|
+          !new_user.public_send(attr).nil? &&
+            (new_user.public_send(attr) != old_user.public_send(attr))
+        end
+        usermod_changes ||= different_groups?(new_user, old_user)
+
+        Yast::Execute.on_target!(USERMOD, *usermod_options(new_user, old_user)) if usermod_changes
+        change_password(new_user, issues) if old_user.password != new_user.password
+        write_auth_keys(new_user, issues) if old_user.authorized_keys != new_user.authorized_keys
+      rescue Cheetah::ExecutionFailed => e
+        issues << Y2Issues::Issue.new(
+          format(_("The user '%{username}' could not be modified"), username: new_user.name)
+        )
+        log.error("Error modifying user '#{new_user.name}' - #{e.message}")
+      end
+
+      # Generates and returns the options expected by `useradd` for given user
+      #
+      # @param user [User]
+      # @return [Array<String>]
+      def useradd_options(user)
+        opts = {
+          "--uid"      => user.uid,
+          "--gid"      => user.gid,
+          "--shell"    => user.shell,
+          "--home-dir" => user.home,
+          "--comment"  => user.gecos.join(","),
+          "--groups"   => useradd_groups(user).join(",")
+        }
+        opts = opts.reject { |_, v| v.to_s.empty? }.flatten
+
+        if user.system?
+          opts << "--system"
+        else
+          opts.concat(create_home_options(user))
+        end
+
+        opts << user.name
+        opts
+      end
+
+      # Name of the secondary groups for useradd
+      #
+      # @return [Array<String>]
+      def useradd_groups(user)
+        names = user.secondary_groups_name
+        return names unless names.empty?
+
+        defaults.secondary_groups
+      end
+
+      # Command to modify the user
+      #
+      # @param new_user [User] User containing the updated information
+      # @param old_user [User] Original user
+      # @return [Array<String>] usermod options
+      # rubocop:disable Metrics/CyclomaticComplexity
+      # rubocop:disable Metrics/AbcSize
+      # rubocop:disable Metrics/PerceivedComplexity
```

Comment:
so much disable raise a bit my internal alarm if we can do it better :)

I see basically two options to reduce its complexity:

1. has method like `modified?(new_user, old_user, :gid)` that can be used in that if ( only exception is gecos which we talk we should change to nil approach if not defined and different_groups? method
2. define map that defines options and conditions. Something like
```ruby
USERMOD_OPTIONS = [
  Proc.new { |o, n| (o.gid != n.gid && n.gid) ?  ["--gid", n.gid] : [] },
  ...
  Proc.new { |o, n| different_groups?(new_user, old_user) ? ["--groups", new_user.secondary_groups_name.join(",")] : []
]
```

But to be honest I do not like neither of it. But I think it missing research did by @dgdavid that said that passing same values are harmless as usermod itself detects it. So maybe condition can be simplified to check for nil.

Reply:
Again, old code. Just copied as-is (including comments... and rubocop comments). Let's take care in a separate card/PR, please.

Reply:
Just for the record, I changed my mind few days after doing such research. I mean, I prefer to provide only modified attributes to the command, no matter what it does currently.